azure/scope/machine.go

```diff
@@ -220,47 +221,109 @@ func (m *MachineScope) InboundNatSpecs(portsInUse map[int32]struct{}) []azure.Re
 
 // NICSpecs returns the network interface specs.
 func (m *MachineScope) NICSpecs() []azure.ResourceSpecGetter {
-	spec := &networkinterfaces.NICSpec{
-		Name:                  azure.GenerateNICName(m.Name()),
-		ResourceGroup:         m.ResourceGroup(),
-		Location:              m.Location(),
-		SubscriptionID:        m.SubscriptionID(),
-		MachineName:           m.Name(),
-		VNetName:              m.Vnet().Name,
-		VNetResourceGroup:     m.Vnet().ResourceGroup,
-		SubnetName:            m.AzureMachine.Spec.SubnetName,
-		AcceleratedNetworking: m.AzureMachine.Spec.AcceleratedNetworking,
-		IPv6Enabled:           m.IsIPv6Enabled(),
-		EnableIPForwarding:    m.AzureMachine.Spec.EnableIPForwarding,
-	}
+	nicSpecs := []azure.ResourceSpecGetter{}
+
+	if len(m.AzureMachine.Spec.NetworkInterfaces) < 1 {
+		spec := &networkinterfaces.NICSpec{
```

Comment:
Can we abstract L227-265 into a convenience fun (e.g., `getDefaultNetworkInterfaceSpec`) so that we can know more easily that if we enter into this flow we will return from `NICSpecs()` immediately after constructing the default NIC object?

```
if len(m.AzureMachine.Spec.NetworkInterfaces) < 1 {
    return append(nicSpecs, getDefaultNetworkInterfaceSpec(m))
}
...
```